src/internal_events/prometheus.rs

```diff
@@ -52,14 +53,20 @@ impl InternalEvent for PrometheusRequestCompleted {
 }
 
 #[derive(Debug)]
-pub struct PrometheusParseError {
+pub struct PrometheusParseError<'a> {
     pub error: ParserError,
     pub url: String,
+    pub body: Cow<'a, str>,
```

Comment:
Even though the source `body` is this type (through `from_utf8_lossy`), you could probably have gotten away with `body: &'a str` here, since `Cow<'a, str>` dereferences to that type. This works, though.